addons/website/controllers/main.py

```diff
@@ -127,6 +124,9 @@ def index(self, **kw):
         def is_reachable(menu):
             return menu.is_visible and menu.url not in ('/', '', '#') and not menu.url.startswith(('/?', '/#', ' '))
 
+        # prefetch all menus (it will prefetch website.page too)
+        top_menu = request.website.menu_id
```

Comment:
This move might be done in a dedicated commit I think, not related to the page caching stuff I think

Reply:
This is related, placed before, it makes an unnecessary query when using the cache (_serve_page).